src/main/java/practice/StreamPractice.java

```diff
@@ -1,80 +1,89 @@
+                .map(n -> n.split(","))
+                .flatMap(Arrays::stream)
+                .map(Integer::parseInt)
+                .mapToInt(Integer::intValue)
+                .filter(n -> n % 2 == 0)
+                .min()
+                .orElseThrow(() -> new RuntimeException("Can't get min value from list" + numbers));
     }
 
-    /**
-     * Given a List of Integer numbers,
-     * return the average of all odd numbers from the list or throw NoSuchElementException.
-     * But before that subtract 1 from each element on an odd position (having the odd index).
-     */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        AtomicInteger index = new AtomicInteger();
+        return numbers.stream()
+                .map(n -> {
+                    int i = index.getAndIncrement();
+                    return i % 2 != 0 ? n - 1 : n;
+                })
+                .filter(n -> n % 2 != 0)
+                .mapToDouble(Double::valueOf)
+                .average()
+                .getAsDouble();
     }
 
-    /**
-     * Given a List of `Person` instances (having `name`, `age` and `sex` fields),
-     * for example, `Arrays.asList( new Person(«Victor», 16, Sex.MAN),
-     * new Person(«Helen», 42, Sex.WOMAN))`,
-     * select from the List only men whose age is from `fromAge` to `toAge` inclusively.
-     * <p>
-     * Example: select men who can be recruited to army (from 18 to 27 years old inclusively).
-     */
     public List<Person> selectMenByAge(List<Person> peopleList, int fromAge, int toAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p -> p.getAge() >= fromAge
+                        && p.getAge() <= toAge
+                        && p.getSex() == Person.Sex.MAN)
+                .collect(Collectors.toList());
     }
 
-    /**
-     * Given a List of `Person` instances (having `name`, `age` and `sex` fields),
-     * for example, `Arrays.asList( new Person(«Victor», 16, Sex.MAN),
-     * new Person(«Helen», 42, Sex.WOMAN))`,
-     * select from the List only people whose age is from `fromAge` and to `maleToAge` (for men)
-     * or to `femaleToAge` (for women) inclusively.
-     * <p>
-     * Example: select people of working age
-     * (from 18 y.o. and to 60 y.o. for men and to 55 y.o. for women inclusively).
-     */
     public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<Person> peopleList) {
-        return Collections.emptyList();
+        Predicate<Person> personPredicate = person -> {
+            if (person.getAge() >= fromAge
+                    && ((person.getSex() == Person.Sex.MAN && person.getAge() <= maleToAge)
+                            || (person.getSex() == Person.Sex.WOMAN
+                            && person.getAge() <= femaleToAge))) {
+                return true;
+            }
+            return false;
```

Comment:
```suggestion
            return person.getAge() >= fromAge
                    && ((person.getSex() == Person.Sex.MAN && person.getAge() <= maleToAge)
                            || (person.getSex() == Person.Sex.WOMAN
                            && person.getAge() <= femaleToAge))
```
Let’s avoid redundant if blocks